resources/shaders/dxr_functions.hlsl

```diff
@@ -33,4 +33,34 @@ float3 unpack_position(float2 uv, float depth, float4x4 inv_vp)
 	return (wpos.xyz / wpos.w).xyz;
 }
 
-#endif //__DXR_FUNCTIONS_HLSL__
\ No newline at end of file
+float3 CalcPeturbedNormal(float3 normal, float3 normal_map, float3 tangent, float3 bitangent, float3 V, out float3 world_normal)
+{
+	float3x4 object_to_world = ObjectToWorld3x4();
+	float3 N = normalize(mul(object_to_world, float4(normal, 0)));
+	float3 T = normalize(mul(object_to_world, float4(tangent, 0)));
+#define CALC_B
```

Comment:
Can this be moved to the top of the file so we can have all configuration properties in an easy-accessible location? Hiding it down here makes it a bit more difficult to control shader behavior.

Reply:
yep